lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemLoader.java

```diff
@@ -211,6 +221,12 @@ private static SdkHttpClient createHttpClient(S3FileSystemConfig config)
         config.getSocketConnectTimeout().ifPresent(timeout -> client.connectionTimeout(timeout.toJavaTime()));
         config.getSocketReadTimeout().ifPresent(timeout -> client.socketTimeout(timeout.toJavaTime()));
 
+        Optional<String> trustStorePath = config.getTruststorePath();
+        Optional<String> trustStorePassword = config.getTruststorePassword();
+        if (trustStorePath.isPresent() && trustStorePassword.isPresent()) {
```

Comment:
```
if (trustStorePath.isPresent()) {
   checkState(trustStorePassword.isPresent(),....
```

Reply:
The similar validation is also within config object - both (truststorePath and truststorePass) set or both not set.